src/elements/stream_element.ts

```diff
@@ -166,22 +166,34 @@ export class StreamElement extends HTMLElement {
   }
 
   private get targetElementsById() {
-    const element = this.ownerDocument?.getElementById(this.target!)
-
-    if (element !== null) {
-      return [element]
-    } else {
-      return []
+    const elementInDom = this.ownerDocument?.getElementById(this.target!)
+    
+    let foundElements : HTMLElement[] = [];
+    
+    if (elementInDom !== null) {
+      foundElements = [elementInDom]
     }
+
+    this.ownerDocument?.querySelectorAll("template").forEach((template) => {
+      let elementInTemplate = template.content.getElementById(this.target!)
+
+      if (elementInTemplate !== null) {
+        foundElements.push(elementInTemplate)
+      }
+    });
+
+    return foundElements;
   }
 
   private get targetElementsByQuery() {
-    const elements = this.ownerDocument?.querySelectorAll(this.targets!)
+    let elements = Array.prototype.slice.call(this.ownerDocument?.querySelectorAll(this.targets!))
```

Comment:
Similar to https://github.com/hotwired/turbo/pull/957/files#r1283567754, we can avoid the `Array.prototype.slice.call` with a `for` loop, and the `Array.prototype.slice.call` with `Array.push` and the `...` spread operator:

```js
const elements = this.ownerDocument?.querySelectorAll(this.targets!)

for (const template of this.ownerDocument?.querySelectorAll("template")) {
  elements.push(...template.content.querySelectorAll(this.targets!))
}

return elements
```

Reply:
Will do that 👍 .